packages/go/cypher/frontend/pattern.go

```diff
@@ -241,81 +206,63 @@ func (s *PatternPredicateVisitor) ExitOC_RelationshipPattern(ctx *parser.OC_Rela
 type PatternVisitor struct {
 	BaseVisitor
 
-	currentPart  *model.PatternPart
 	PatternParts []*model.PatternPart
 }
 
-func (s *PatternVisitor) EnterOC_AnonymousPatternPart(ctx *parser.OC_AnonymousPatternPartContext) {
```

Comment:
PatternPart has been moved to its own visitor since we needed it for MERGE. PatternVisitor now correctly defers to PatternPart, so all the Pattern specific functions could be safely removed. Note that we also removed EnterOC_RelationshipsPattern and ExitOC_RelationshipsPattern, due to them having nothing to do with Pattern/PattenPart (they do call into some similar stuff, but they do not contain PatternParts, so the implementation was incorrect anyway)